### PR TITLE
[Serialization] Remove "delayed actions" support

### PR DESCRIPTION
**Pull request by jrose-apple**
Previously, cycle-breaking logic would delay certain actions until all re-entrant deserialization was complete for a particular module. That hasn't been used in a while, though, now that the AST itself supports more laziness, so let's take it out.

No functionality change; this really was unused. PR also contains some other minor cleanups in deserialization.

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please test

File: lib/Serialization/Deserialization.cpp [hunk 1]
```diff
@@ -143,14 +143,9 @@ static void skipRecord(llvm::BitstreamCursor &cursor, unsigned recordKind) {
   auto next = cursor.advance(AF_DontPopBlockAtEnd);
   assert(next.Kind == llvm::BitstreamEntry::Record);
 
-#if NDEBUG
-  cursor.skipRecord(next.ID);
-#else
-  SmallVector<uint64_t, 64> scratch;
-  StringRef blobData;
-  unsigned kind = cursor.readRecord(next.ID, scratch, &blobData);
+  unsigned kind = cursor.skipRecord(next.ID);
```
Comment (brentdax) [inline lib/Serialization/Deserialization.cpp]: I assume there was an upstream API change to return the `kind` from `skipRecord()` that we hadn't adopted until now?

**Review by beccadax — APPROVED** [review]

**Comment by slavapestov** [comment]
Nice cleanup!

File: lib/Serialization/Deserialization.cpp (hunk 1 above)
Comment (jrose-apple) [reply lib/Serialization/Deserialization.cpp]: Presumably! I didn't bother to spelunk through LLVM's history in this case, only saw that we can do better.

**Pull request merged**